src/legacy/core_plugins/kibana/public/dashboard/top_nav/options.tsx

```diff
@@ -18,40 +18,47 @@
  */
 
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { injectI18n } from '@kbn/i18n/react';
+import { injectI18n, InjectedIntl } from '@kbn/i18n/react';
 
-import {
-  EuiForm,
-  EuiFormRow,
-  EuiSwitch,
-} from '@elastic/eui';
+import { EuiForm, EuiFormRow, EuiSwitch } from '@elastic/eui';
 
-class OptionsMenuUi extends Component {
+interface Props {
+  useMargins: boolean;
+  onUseMarginsChange: (useMargins: boolean) => void;
+  hidePanelTitles: boolean;
+  onHidePanelTitlesChange: (hideTitles: boolean) => void;
+  intl: InjectedIntl;
+}
+
+interface State {
+  useMargins: boolean;
+  hidePanelTitles: boolean;
+}
 
-  state = {
-    useMargins: this.props.useMargins,
-    hidePanelTitles: this.props.hidePanelTitles,
+class OptionsMenuUi extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
```

Comment:
I think you can leave this outside of constructor, like in `.js` file.

![image](https://user-images.githubusercontent.com/9773803/59513228-b435d980-8eba-11e9-9fa0-7d5a11fb731f.png)